dali/pipeline/operators/op_spec.h

```diff
@@ -445,7 +490,12 @@ inline T OpSpec::GetArgumentImpl(
   // Search for the argument in tensor arguments first
   if (this->HasTensorArgument(name)) {
     DALI_ENFORCE(ws != nullptr, "Tensor value is unexpected for argument \"" + name + "\".");
-    const auto& value = ws->ArgumentInput(name);
+    const auto &value = ws->ArgumentInput(name);
+    DALI_ENFORCE(kernels::is_uniform(value.shape()),
+                 "Arguments should be passed as uniform TensorLists. Argument \"" + name +
```

Comment:
Aren't non-uniform argument inputs legal when obtained via `ArgumentInput`? This error suggests they aren't.

Reply:
Also, the check for uniformity could be inside CheckArgumentShape - now it's not checked in TryGetArgumentImpl.

Reply:
Still, this check is not used with ArgumentInput but only with Arguments.